workflow.md

```diff
@@ -5,9 +5,218 @@ title: Samtools - Workflows
 
 # Workflows
 
+* [FASTQ to BAM/CRAM processing](#fastq_to_bam)
 * [WES Mapping to Variant Calls - Version 1.0](#mapping_to_variant)
 * [Using CRAM within Samtools](#mapping_to_cram)
 
+## <a name="fastq_to_bam"></a>FASTQ to BAM / CRAM - Version 1.0
+
+The sequencing instruments produce unaligned data, typically in FASTQ
+format.  It is possible to store unaligned data in BAM or CRAM, and
+indeed it may be preferable as it permits meta-data in the header and
+per-record auxiliary tags, however in this workflow we consider the
+end product to be a sorted aligned BAM or CRAM file so we cover that
+at the end.
+
+There are two primary ways of producing this:
+
+* Alignment / mapping to a known reference
+* De-novo assembly
+
+#### Alignment / mapping to a known reference
+
+There are many tools for mapping sequences (reads) against a reference
+file.  We do not recommend any single tool over another, but for
+illustrative purposes are using [Minimap2](https://github.com/lh3/minimap2).
+
+There are processing steps which together make a pipeline:
+
+* Map / align
+* Fix mate-pair issues
+* Mark duplicates (part 1: preparation)
+* Sort to positional order
+* Mark duplicates (part 2: marking)
+* Convert to final file format
+
+We initially break these down into individual stages, reading and
+writing from disk, but it is worth considering using a UNIX pipeline
+to link these together avoiding temporary files.
+
+#### Mapping
+
+A basic mapping tool may take a single or pair of FASTQ files and
+align against either a pre-built index or directly against a fasta file
+(as used below).  The most naive approach is simply to save this file
+as SAM:
+
+    minimap2 -t 8 -a -x sr C.Elegans.fa SRR065390_1.fastq SRR065390_2.fastq > CE.sam
```

Comment:
Maybe show  it with `-o CE.sam` instead of redirection, to fend off the usual accidentally-capturing-stderr problems when people use this with `nohup` or their queuing systems.

Reply:
I used ">" because I couldn't find an output option.   I've looked again and it's definitely not there.  The man page recommends `>` too.

Good point about nohup though.  I'll add a comment about this.

Reply:
Sigh.  Typical Sanger oddity then.  I looked at our usual team software dir and on Jan 20th we installed minimap 2.10.  I naively assumed it was current!

Looking harder at other team dirs I've now found a grand total of 23 copies, all bar 2 of them 2.17.  I chose... poorly!

Reply:
Well, it was put there by conda, so who knows ;-)

(To be fair Conda install 2.17 for me, but I don't know what else was in that environment.)

Reply:
Anyway, I updated this PR but kept the nohup comment in there as it's generally a useful observation and has been the cause of several cryptic SAM parsing errors in the past.

Reply:
You people are actually **using** conda? Even I'm not that crazy!

Reply:
Some groups even package up all their own internal software via conda in order to use it as a system for managing dependencies.  (!)